DNN Platform/Library/Entities/Modules/Settings/SettingsRepository.cs

```diff
@@ -63,23 +63,16 @@ public void SaveSettings(ModuleInfo moduleContext, T settings)
 
                         if(string.IsNullOrEmpty(settingValueAsString))
                         {
-                            if (settingValue is DateTime)
-                            {
-                                settingValueAsString = ((DateTime)settingValue).ToString("u");
-                            }
-                            else
-                            {
-                                settingValueAsString = settingValue.ToString();
-                            }
+                            settingValueAsString = GetSettingValueAsString(settingValue);
```

Comment:
Should this just call `Convert.ToString(settingValue, CultureInfo.InvariantCulture)`, to cover every type that implements `IConvertible` or `IFormattable`?  The only difference that I see is that the date would be formatted differently, but so long as it's consistently the same culture, it shouldn't matter, right?